sdcm/utils/latency.py

```diff
@@ -0,0 +1,115 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2020 ScyllaDB
+
+import time
+from sdcm.db_stats import PrometheusDBStats
+
+
+def avg(values):
+    if values:
+        return float(format(sum([float(val) for val in values])/len(values), '.1f'))
+    else:
+        return None
+
+
+def get_node_by_ip(cluster, ip):
+    for node in cluster.nodes():
+        if ip == node.ip_address:
+            return node
+
+
+def collect_latency(monitor_node, start, end, load_type, cluster):
+    res = dict()
+    prometheus = PrometheusDBStats(host=monitor_node.external_address)
+    duration = int(end - start)
+    cassandra_stress_precision = ['99', '95']  # in the future should include also 'max'
+    scylla_precision = ['99']  # in the future should include also '95', '5'
+
+    for precision in cassandra_stress_precision:
+        metric = f'c-s {precision}' if precision == 'max' else f'c-s P{precision}'
+        if not precision == 'max':
+            precision = f'perc_{precision}'
+        query = f'collectd_cassandra_stress_{load_type}_gauge{{type="lat_{precision}"}}'
+        query_res = prometheus.query(query, start, end)
+        lst = list()
+        for entry in query_res:
+            response = None
```

Comment:
not clear names `lst` and `response`. May be latency_values_lst and lat_value?

Reply:
updated